source/js/components/DatePicker.tsx

```diff
@@ -23,10 +23,9 @@ const DatePicker = ({ availableDates, date, required, onDateSelected }: DatePick
   };
 
   const formatTime = (timeSlot: TimeSlot) =>
-    `${timeSlot.startTime?.substring(0, timeSlot.startTime.indexOf('+'))}-${timeSlot.endTime?.substring(
-      0,
-      timeSlot.startTime?.indexOf('+'),
-    )}`;
+    `${new Date(`${timeSlot.date}T${timeSlot.startTime}`).toLocaleTimeString()}-${new Date(
+      `${timeSlot.date}T${timeSlot.endTime}`,
+    ).toLocaleTimeString()}`;
```

Comment:
**OPTIONAL**
The creation of localTimeString could be done in a separate function.

```js
const toLocaleTimeString = (date: string) => {
  return new Date(date).toLocaleTimeString();
}

const formatTime = (timeSlot: TimeSlot) => {
 const startTimeString = toLocaleTimeString(`${timeSlot.date}T${timeSlot.startTime}`);
 const endTimeString = toLocaleTimeString(`${timeSlot.date}T${timeSlot.endTime}`);

 return `${startTimeString}-${endTimeString}`;
}
```